src/main/common/utils.js

```diff
@@ -190,6 +190,78 @@ gpii.app.isPointInRect = function (point, rectangle) {
            rectangle.y <= point.y && point.y <= rectangle.y + rectangle.height;
 };
 
+/**
+ * Checks if the buttonList attribute exists in the siteConfig object
+ * @param {Object} siteConfig - instance of the siteConfig object
+ * @returns {Boolean}
+ */
+gpii.app.hasButtonList = function(siteConfig) {
+    return fluid.isValue(siteConfig.buttonList);
+};
+
+/**
+ * Looks for a `id` and matches it to the provided string
+ * return empty array when there is no button found
+ * @param {String} buttonId - the `id` of the button
+ * @param {Object[]} availableButtons - the full settings list
+ * @returns {Boolean|Object} - returns false when button is not found,
+ * or the setting object if found
+ */
+gpii.app.findButtonById = function(buttonId, availableButtons) {
+    if (fluid.isValue(buttonId) && fluid.isValue(availableButtons)) {
+        return fluid.find_if(availableButtons, function(button) {
+            if (fluid.isValue(button.id)) {
```

Comment:
This guard can be removed since you already checked that the buttonId is a value

Reply:
The check above (line 211) is for the param buttonId, the one here (line 213) checks if the current button in the foreach has `id` before it compares it to the buttonId itself.

The `id` node its not required, and a few buttons don't have one (all of the service buttons for example), so the check on line 214 throws errors when encounters one.

Reply:
It could never throw an error, since the JS === operator just works by reference equality. Try removing the if statement and see what happens :)
Also, please run the branch through the linter - cheers

Reply:
OK, OK, let's say that I agree 😸 . All of the changes are up, including the lint ones.